lib/Config/Hooks.php

```diff
@@ -206,6 +213,14 @@ function setupWooCommerceUsers() {
     );
   }
 
+
+  function setupWooCommerceOrders() {
+    $this->wp->addAction(
+      'woocommerce_payment_complete',
```

Comment:
Does @costasovo know which one is the correct one? From https://docs.woocommerce.com/wc-apidocs/source-class-WC_Order.html#327 and https://docs.woocommerce.com/document/managing-orders/ it seems to me I could use `woocommerce_order_status_ processing` and `woocommerce_order_status_completed` (both since the logic is "createOrUpdate" so it won't track anything twice).

Reply:
I think we should use both. I think that there could be an order paid using `cheque` and such order transitions to status `on hold` and some admin may switch it to `completed` and it would skip `processing`.
See how it is done in premium https://github.com/mailpoet/mailpoet-premium/blob/master/lib/AutomaticEmails/WooCommerce/Events/FirstPurchase.php#L43

We use filter there because WC allows the user to modify how orders states are changed so the filter is a possibility to alter states we are using to match his altered state transitions. It is not documented though :(

Reply:
@costasovo Wow, thanks for explanation. So I should run it through the same filter, right? Will fix.

Reply:
The filter in premium has specific name for the event so maybe this one should be also extra filter. Not sure...